File: Flyweight/Line.cs
namespace Flyweight
{
    internal class Line : IShape
    {
        public void Draw(Graphics g, int x1, int y1, int x2, int y2, Color color, TextBox outputTextBox, IShape shape)
        {
            DrawingForm.Instance.Invoke((MethodInvoker)(() => DrawingForm.Instance.ClearMessage()));
            DrawingForm.Instance.Invoke((MethodInvoker)(() => DrawingForm.Instance.ShowMessage("Creating " + shape.GetType().Name + "...")));
            Thread.Sleep(2000);
            g.DrawLine(new Pen(color), x1, y1, x2, y2);
        }
    }
}

File: Java/Flyweight/src/Line.java
import java.awt.*;

class Line implements IShape {
    private final int x1;
    private final int y1;
    private final int x2;
    private final int y2;
    private final Color color;

    public Line(int x1, int y1, int x2, int y2, Color color) {
        this.x1 = x1;
        this.y1 = y1;
        this.x2 = x2;
        this.y2 = y2;
        this.color = color;
    }

    @Override
    public void draw(Graphics g) {
        DrawingForm.Instance.clearMessage();
        DrawingForm.Instance.showMessage("Creating Line...");
        g.setColor(color);
        g.drawLine(x1, y1, x2, y2);
    }
}